### PR TITLE
feat: Add confirm dialogs when toggling dangerous settings

### PR DESCRIPTION
**Pull request by TheAabedKhan**
Note: Strings need to be set

### CODE REVIEW

File: app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt [hunk 1]
```diff
@@ -5,27 +5,53 @@ import androidx.compose.foundation.clickable
...
     BooleanItem(
         modifier = modifier,
         value = value,
-        onValueChange = { coroutineScope.launch { preference.update(it) } },
+        onValueChange = {
+            if (isSafeguard && it != suggestedValue) {
+                showSafeguardWarning = true
+            } else {
+                coroutineScope.launch { preference.update(it) }
+            }},
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt:54]: Add a newline here

File: app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt [hunk 2]
```diff
@@ -5,27 +5,53 @@ import androidx.compose.foundation.clickable
...
 import androidx.compose.ui.res.stringResource
 import app.revanced.manager.domain.manager.base.Preference
+import app.revanced.manager.ui.component.SafeguardConfirmationDialog
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
 
 @Composable
 fun BooleanItem(
     modifier: Modifier = Modifier,
+    isSafeguard: Boolean = false,
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt:22]: This parameter can be removed if you rename the overload to `SafeguardBooleanItem` and copy the original function from the `compose-dev` branch

File: app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt [hunk 3]
```diff
@@ -5,27 +5,53 @@ import androidx.compose.foundation.clickable
...
+    suggestedValue: Boolean = false,
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt:23]: You can get this from the `preference` if you make `Preference.default` public

File: app/src/main/java/app/revanced/manager/ui/component/SafeguardConfirmationDialog.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import app.revanced.manager.R
+
+@Composable
+fun SafeguardConfirmationDialog(
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/component/SafeguardConfirmationDialog.kt:16]: Move this file to `ui/component/settings`

**Review by Axelen123 — CHANGES_REQUESTED** [review]

File: app/src/main/java/app/revanced/manager/ui/component/settings/SafeguardBooleanItem.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import kotlinx.coroutines.launch
+
+@Composable
+fun SafeguardBooleanItem(
+    modifier: Modifier = Modifier,
+    preference: Preference<Boolean>,
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
+    @StringRes headline: Int,
+    @StringRes description: Int,
+    @StringRes confirmationText: Int = 0
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/component/settings/SafeguardBooleanItem.kt:23]:
```suggestion
    @StringRes confirmationText: Int
```

**Review by Axelen123 — CHANGES_REQUESTED** [review]

**Review by Axelen123 — APPROVED** [review]
The UI code looks good to me

**Review by validcube — CHANGES_REQUESTED** [review]
🔥 The dialog's title & description for safeguard must tell the user what the feature does and the consequences of enabling/disabling it.

* (consequences): https://m3.material.io/foundations/content-design/style-guide/ux-writing-best-practices#d7f04666-89f8-4795-a563-16d060b5ccf2
* (_not really important that much_, title): https://m3.material.io/components/dialogs/guidelines#0cff8b6b-d4f3-4442-8194-31212f5d1a12

image

Additionally, this isn't very clear to the user, 

"Enforce selection of the suggested app version"
* Yes (Toggle the preference - which is on by default)
* No (Dismiss the dialog)

~~I assume you already knew that, so this is just a red flag check to tell maintainer to not accidentally merge.~~

**Comment by Axelen123** [comment]
Are there strings in flutter manager that could be used for this?

**Comment by TheAabedKhan** [comment]
> Are there strings in flutter manager that could be used for this?

We have these two in flutter

```json
{
"enablePatchesSelectionWarningText": "Changing the selection of patches may cause unexpected issues.\n\nEnable anyways?",
"requireSuggestedAppVersionDialogText": "Selecting an app that is not the suggested version may cause unexpected issues.\n\nDo you want to proceed anyways?"
}
```

**Comment by Axelen123** [comment]
I guess those strings can be reused

**Pull request merged**